src/lib/core/hooks/analysis.ts

```diff
@@ -61,8 +61,14 @@ const analysisCache: Record<string, Analysis | undefined> = {};
 
 export function usePreloadAnalysis() {
   const analysisClient = useAnalysisClient();
-  return async function preloadAnalysis(id: string) {
-    analysisCache[id] = await analysisClient.getAnalysis(id);
+  /**
+   * @param id Identifier of saved analysis.
+   * @param analysis Optional analysis object to use for the cache. This can be
+   * used if an analysis object was already fetched from the service. This
+   * should be used sparingly.
+   */
+  return async function preloadAnalysis(id: string, analysis?: Analysis) {
+    analysisCache[id] = analysis ?? (await analysisClient.getAnalysis(id));
   };
```

Comment:
Since `preloadAnalysis` with a passed `analysis` doesn't actually "preload" anything, it might be clearer to make a dedicated `cacheAnalysis` utility which receives an `id` and `analysis` object, and simply stores that `analysis` in the cache. You could then: 

1. Refactor `preloadAnalysis` to use `cacheAnalysis`, and don't allow an optional `analysis` argument
2. In the `useWorkspaceAnalysis` hook, directly use `cacheAnalysis` instead of `preloadAnalysis`

If you do this, I would add a comment that `cacheAnalysis` should be used sparingly, similar to what you did for `preloadAnalysis`.

```suggestion
  return async function preloadAnalysis(id: string) {
    cacheAnalysis(id, await analysisClient.getAnalysis(id));
  };
```

Reply:
This makes a lot of sense. I was thinking that the optional argument meant I was overloading the semantics of the function :slightly_smiling_face: